sorts/merge_sort.ts

```diff
@@ -29,14 +29,15 @@ function merge (array1: number[], array2: number[]): number[] {
   const result = []
   let index1 = 0
   let index2 = 0
-  
+
   while (index1 < array1.length && index2 < array2.length) {
     if (array1[index1] < array2[index2]) {
       result.push(array1[index1++])
     } else {
       result.push(array2[index2++])
     }
   }
-  
-  return [...result, ...array1.slice(index1), ...array2.slice(index2)]
+
+  return result.concat(array1.slice(index1)).concat(array2.slice(index2))
```

Comment:
This is the same thing but more verbose. It still copies the arrays. It still doesn't properly allocate result as the previous code did.

Reply:
Yeah, you're right 😂
I have no idea. Do you have any suggestions? or maybe I should close it for good 😅
thank you time anyway.